### PR TITLE
[GEOT-7341] Update Spotbugs to 4.7.3

### PR DESCRIPTION
**Pull request by aaime**
[GEOT-7341](https://osgeo-org.atlassian.net/browse/GEOT-7341) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=geotools&utm_campaign=badge_info)

See ticket

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [x] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [ ] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [x] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [x] Bug fixes and small new features are presented as a single commit.
- [x] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

**Review by jodygarnett — APPROVED** [review]
Thanks for the upgrade!

File: modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/Contours.java [hunk 1]
```diff
@@ -45,6 +45,11 @@
 
 public class Contours {
 
+    private enum PointType {
+        Point,
+        MultiPoint
```
Comment (jodygarnett) [inline modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/Contours.java:50]: Is this change related? Or off topic ...

Comment (sikeoka) [reply modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/Contours.java:50]: I think this is related to a rule that looks for using == to compare strings instead of the equals method.  It seems like a simple boolean flag could be used here instead of strings or an enum.

File: modules/library/referencing/src/main/java/org/geotools/referencing/crs/DefaultProjectedCRS.java [hunk 2]
```diff
@@ -250,6 +251,7 @@ public int hashCode() {
      * @return The name of the WKT element type, which is {@code "PROJCS"}.
      */
     @Override
+    @SuppressFBWarnings("ES_COMPARING_STRINGS_WITH_EQ")
```
Comment (sikeoka) [inline modules/library/referencing/src/main/java/org/geotools/referencing/crs/DefaultProjectedCRS.java:254]: Wouldn't it be better to change x==y to x.equals(y) to fix the warning even if == is okay here instead of suppressing the warning?

Comment (jodygarnett) [reply modules/library/referencing/src/main/java/org/geotools/referencing/crs/DefaultProjectedCRS.java:254]: Using ``equals`` would be a performance difference; not sure how performance sensitive this is.

File: modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/Contours.java (hunk 1 above)
Comment (aaime) [reply modules/unsupported/process-feature/src/main/java/org/geotools/process/vector/Contours.java:50]: Yep, it's to avoid string comparison with identity equality.

File: modules/library/referencing/src/main/java/org/geotools/referencing/crs/DefaultProjectedCRS.java (hunk 2 above)
Comment (aaime) [reply modules/library/referencing/src/main/java/org/geotools/referencing/crs/DefaultProjectedCRS.java:254]: I left it with == since the original author went to the trouble of writing a comment explaining that identity comparison was ok in that case. Also, code is convoluted (name is assigned during a if and result passed into a method as a side effect... horrible...)
